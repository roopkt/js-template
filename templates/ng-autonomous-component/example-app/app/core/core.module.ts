import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { AppComponent } from './containers/app.component';
import { MaterialModule } from '../material';

export const COMPONENTS = [AppComponent];

@NgModule({
  imports: [CommonModule, MaterialModule, RouterModule],
  declarations: COMPONENTS,
  exports: COMPONENTS,
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
      providers: [],
    };
  }
}
