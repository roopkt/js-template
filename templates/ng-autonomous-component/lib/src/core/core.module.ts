import { CommonModule } from '@angular/common';
import { ModuleWithProviders, NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

export const COMPONENTS: any[] = [];

@NgModule({
  imports: [CommonModule, RouterModule],
  declarations: COMPONENTS,
  exports: COMPONENTS,
})
export class CoreModule {
  static forFeature(): ModuleWithProviders {
    return {
      ngModule: CoreModule,
      providers: [],
    };
  }
}
