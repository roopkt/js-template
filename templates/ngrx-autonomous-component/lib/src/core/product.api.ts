import { Injectable, Inject } from '@angular/core';
import { Product } from '../models/product';
import { Observable } from 'rxjs/Observable';
import { of } from 'rxjs/observable/of';
import { FEATURE_MODULE_CONFIG, FeatureModuleConfig } from '../config';

@Injectable()
export class ProductApi {

  constructor(@Inject(FEATURE_MODULE_CONFIG) private config: FeatureModuleConfig) {
  }

  createProduct(product: Product): Observable<Product> {
    // make actual server side call using baseurl provided from config
    return of(product);
  }
}
