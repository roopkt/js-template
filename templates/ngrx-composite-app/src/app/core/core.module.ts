import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { LayoutComponent } from './components/layout.component';
import { NavItemComponent } from './components/nav-item.component';
import { AppComponent } from './containers/app.component';
import { FooterComponent } from './components/footer.component';
import { HeaderComponent } from './components/header.component';
import { BodyComponent } from './components/body.component';
import { ProgressBarContainerComponent } from './containers/progress-bar-container.component';
import { ProgressBarComponent } from './components/progress-bar.component';

export const COMPONENTS = [
  AppComponent,
  LayoutComponent,
  NavItemComponent,
  FooterComponent,
  HeaderComponent,
  BodyComponent,
  ProgressBarComponent,
  ProgressBarContainerComponent
];

@NgModule({
  imports: [CommonModule, RouterModule],
  declarations: COMPONENTS,
  exports: COMPONENTS,
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
      providers: [],
    };
  }
}
