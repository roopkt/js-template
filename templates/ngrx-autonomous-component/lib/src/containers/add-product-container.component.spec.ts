import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AddProductContainerComponent } from './add-product-container.component';
import { ReactiveFormsModule } from '@angular/forms';

// todo fix test
xdescribe('Add Product Page', () => {
  let fixture: ComponentFixture<AddProductContainerComponent>;
  let component: AddProductContainerComponent;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [ReactiveFormsModule],
        declarations: [AddProductContainerComponent],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(AddProductContainerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should compile', () => {
    expect(component).toBeTruthy();
  });
});

