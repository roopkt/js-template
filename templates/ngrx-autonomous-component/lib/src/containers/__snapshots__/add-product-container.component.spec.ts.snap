// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Product Page should compile 1`] = `undefined`;
